File: 690 - Employee Importance.java
/*
// Definition for Employee.
class Employee {
    public int id;
    public int importance;
    public List<Integer> subordinates;
};
*/

class Solution {
    public int getImportance(List<Employee> employees, int id) {
        int sum = 0;
        for (Employee i : employees) {
            if (i.id == id) {
                sum = i.importance;
                for (int subid : i.subordinates) {
                    sum += getImportance(employees, subid);
                }
                break;
            }
        }
        return sum;
    }
}

File: 690 - Employee Importance.js
/**
 * Definition for Employee.
 * function Employee(id, importance, subordinates) {
 *     this.id = id;
 *     this.importance = importance;
 *     this.subordinates = subordinates;
 * }
 */

/**
 * @param {Employee[]} employees
 * @param {number} id
 * @return {number}
 */
var GetImportance = function (employees, id) {
    let sum = 0;
    for (let emp of employees) {
        if (emp.id == id) {
            sum = emp.importance;
            for (let subid of emp.subordinates) {
                sum += GetImportance(employees, subid);
            }
            break;
        }

    }
    return sum;
};